test/browser/time-picker/time-picker-spec.js

```diff
@@ -56,7 +64,31 @@ describe('TimePicker', () => {
     );
 
     const timeTextField = TestUtils.findRenderedComponentWithType(render, TextField);
+    const timepicker = TestUtils.findRenderedComponentWithType(render, TimePicker);
+    const formattedTime = formatTime(valueTime);
 
-    expect(timeTextField.props.value, formatTime(valueTime));
+    expect(timepicker.state.time).to.be.equal(valueTime);
+    expect(formatTime(timepicker.state.time)).to.be.equal(formattedTime);
+    expect(timeTextField.props.value).to.be.equal(formattedTime);
+  });
+
+  it('should have time null if value is null', () => {
+    const valueTime = null;
+
+    const render = TestUtils.renderIntoDocument(
+      <ThemedTimePicker
+        value={valueTime}
+        format="ampm"
+        locale="en-US"
+      />
+    );
+
+    const timeTextField = TestUtils.findRenderedComponentWithType(render, TextField);
+    const timepicker = TestUtils.findRenderedComponentWithType(render, TimePicker);
+    const formattedTime = formatTime(valueTime);
+
+    expect(timepicker.state.time).to.be.equal(valueTime);
```

Comment:
@oliviertassinari regarding your comment at: https://github.com/callemall/material-ui/pull/3993#discussion_r59919360, in theory this test covers that case, if null is passed the `state.time` is null too